common/v2/services/Store/StoreProvider.tsx

```diff
@@ -59,6 +70,67 @@ export const StoreProvider = ({ children }: { children: React.ReactNode }) => {
     [accounts]
   );
 
+  useEffect(() => {
+    setPendingTransactions(
+      accounts
+        .map(account =>
+          account.transactions.map(tx => {
+            return { ...tx, network: account.network };
+          })
+        )
+        .flatMap(tx => tx)
+        .filter((txObject: ITxReceipt) => txObject.stage === ITxStatus.PENDING) || []
+    );
+  }, [accounts]);
+
+  // A pending transaction is detected.
+  useEffect(() => {
+    if (pendingTransactions.length <= 0) {
+      return;
+    }
+    // This interval is used to poll for status of txs.
+    const blockNumInterval = setInterval(() => {
+      pendingTransactions.forEach((pendingTransactionObj: any) => {
```

Comment:
This is a good opportunity to apply the aforementioned coding principles such as succinct naming, destructuring, separation of concerns. To go further you can have fun with a Promise returning [pipe](https://ramdajs.com/0.19.1/docs/#pipeP)!